app/scripts/services/quota.js

```diff
@@ -254,6 +263,100 @@ angular.module("openshiftConsole")
+      var msg;
+      if (_.includes(COMPUTE_RESOURCE_QUOTAS, quotaKey)) {
+        msg = msgPrefix + "It is using " + percent((usedValue/hardValue), 0) + " of " + usageWithUnits(hard, quotaKey) + " " + humanizeQuotaResource(quotaKey, true) + ".";
+      } else {
+        msg = msgPrefix + "It is using " + usedValue + " of " + hardValue + " " + humanizeQuotaResource(quotaKey, true) + ".";
+      }
+
+      if (Constants.QUOTA_NOTIFICATION_MESSAGE && Constants.QUOTA_NOTIFICATION_MESSAGE[quotaKey]) {
+        msg += " " + Constants.QUOTA_NOTIFICATION_MESSAGE[quotaKey];
+      }
+
+      return msg;
+    };
+
+    // Return notifications if you are at quota or over any quota for any resource. Do *not*
+    // warn about quota for 'resourcequotas' or resources whose hard limit is
+    // 0, however.
+    var getQuotaNotifications = function(quotas, clusterQuotas, projectName) {
+      var notifications = [];
+
+      var notificationsForQuota = function(quota) {
+        var q = quota.status.total || quota.status;
+        _.each(q.hard, function(hard, quotaKey) {
+          var hardValue = usageValue(hard);
+          var used = _.get(q, ['used', quotaKey]);
+          var usedValue = usageValue(used);
+
+          // We always ignore quota warnings about being out of
+          // resourcequotas since end users cant do anything about it
+          if (quotaKey === 'resourcequotas' || !hardValue || !usedValue) {
+            return;
+          }
+
+          if(hardValue <= usedValue) {
+            var msg = getNotificaitonMessage(used, usedValue, hard, hardValue, quotaKey);
+            var containsHTML = hasHTML(msg);
+            notifications.push({
+              id: "quota-limit-reached-" + quotaKey,
+              namespace: projectName,
+              type: (hardValue < usedValue ? 'warning' : 'info'),
+              message: msg,
+              isHTML: containsHTML,
+              skipToast: true,
+              showInDrawer: true,
+              actions: [
+                {
+                  name: 'View Quotas',
+                  title: 'View project quotas',
+                  onClick: function(notification, action, index, group, drawer) {
+                    $location.url("/project/" + $routeParams.project + "/quota");
+                    drawer.drawerHidden = true;
+                  }
+                },
+                {
+                  name: "Don't Show Me Again",
+                  title: 'Permenantly hide this notificaiton until quota limit changes',
+                  onClick: function(notification, action, index, group, drawer) {
+                    NotificationsService.permanentlyHideNotification(notification.uid, notification.namespace);
+                    EventsService.markCleared(notification.uid);
+                    group.notifications.splice(index, 1);
+                    drawer.customScope.countUnreadNotifications();
```

Comment:
I'd change these onClick handlers to `return true` if they should clear the notification and then move these two lines back to the drawer. The `onClick` handler shouldn't need to take any parameters (you have a reference to the notification already through the closure).

Reply:
Notification drawer wrapper can check the onClick return value and then clear the notification if it's true.

Reply:
`action.onClick` isn't being called from the notificationDrawerWrapper.js, it is being called from the included `notification-body.html`, [here](https://github.com/openshift/origin-web-console/pull/2116/files#diff-900a7b20a52c72356948916b917b0ebdR39).  
In `quota.js`, I could throw `$rootScope.$emit('NotificationDrawerWrapper.clear')` and also pass along the notification.id/uid.

Reply:
> action.onClick isn't being called from the notificationDrawerWrapper.js, it is being called from the included notification-body.html

Right, I understand. I'm proposing changing that to follow this basic pattern:

https://github.com/openshift/origin-web-console/blob/master/app/views/_alerts.html#L24

https://github.com/openshift/origin-web-console/blob/master/app/scripts/directives/alerts.js#L34-L42